terraform/deployments/ga4-analytics/service_account.tf

```
@@ -0,0 +1,25 @@
+resource "google_service_account" "sa--ga4-analytics-352613" {
```

```suggestion
resource "google_service_account" "ga4-analytics" {
```

Avoiding naming with arbitrary descriptors (we don't need to know the project id in the tf resource name)